File: elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  require ResistorColor

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    [c1 | rest] = colors
    [c2 | _] = rest
    10 * ResistorColor.code(c1) + ResistorColor.code(c2)
  end
end

File: go/resistor-color-trio/lib/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

import "resistorcolor"

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
    band1 := resistorcolor.ColorCode(colors[0])
    band2 := resistorcolor.ColorCode(colors[1])
	return 10 * band1 + band2
}
